### ISSUE TITLE
Accuracy of FocalLoss in image classification was very low

### ISSUE THREAD

**Issue by lixiangchun** [dmlc/gluon-cv#866]
In my image classification situation where the positive samples and negative samples are unbalanced (i.e. 1:10), I replaced `gluoncv.loss.FocalLoss` with `gluon.loss.SoftmaxCrossEntropyLoss` in a finetuning task, which is to differentiate between the positive and negative samples. For `FocalLoss`, the error rate on the validation set is very high:

```
[Epoch 0] validation: err-top1=0.467000
[Epoch 1] validation: err-top1=0.511000
[Epoch 2] validation: err-top1=0.530000
[Epoch 3] validation: err-top1=0.526000
[Epoch 4] validation: err-top1=0.535000
[Epoch 5] validation: err-top1=0.529000
```

For `SoftmaxCrossEntropyLoss`, the error rate on the validation set is quite low:
```
[Epoch 0] validation: err-top1=0.087000
[Epoch 1] validation: err-top1=0.072000
[Epoch 2] validation: err-top1=0.060000
[Epoch 3] validation: err-top1=0.057000
[Epoch 4] validation: err-top1=0.060000
[Epoch 5] validation: err-top1=0.046000
```

**Comment by hetong007** [comment]
@zhanghang1989 would you please help?

**Comment by zhanghang1989** [comment]
Focal loss is similar to label smoothing, but this implementation is designed for binary prediction. You can change it for your need.

https://github.com/dmlc/gluon-cv/blob/master/gluoncv/loss.py#L77

One solution is add softmax at the end of your model and change `from_logits=True`
Or you can change `pred = F.sigmoid(pred)` to `pred = F.log_softmax(pred, self._axis)`

**Comment by lixiangchun** [comment]
Thanks for your help @zhanghang1989 @hetong007 . The solution works, however, the top1 error is still higher for `FocalLoss` versus `SoftmaxCrossEntropyLoss` (13.3% versus 4.6%). 

Do you have any suggestion how to deal with extremely imbalanced image classification besides data augmentation, label smoothing, and mixup training? Imbalanced classification is quite common when analyzing medical imaging data.

**Comment by hetong007** [comment]
@lixiangchun there's one paper which we haven't tested: https://arxiv.org/abs/1901.05555. It is published on CVPR2019 and you may give it a try.

**Comment by lixiangchun** [comment]
@hetong007 thanks for your message and great work. I will read the paper and have a try. BTW, look forward to the official implementation of class-balanced loss in `gluoncv`.

**Comment by StacyYang** [comment]
Besides Tong’s paper. You may change alpha to change the strength of “label smoothing” in the Focal Loss. Or use OHEM loss.

**Comment by lixiangchun** [comment]
@StacyYang Thanks for your comments.
